### DISCUSSION TITLE
Berry - Convert symbol to Berry escape format

### DISCUSSION THREAD

**Discussion by valdus2** [arendst/Tasmota#22691] [Q&A] [answered]
I use the typicons font in Haspmota, and I'm trying to change symbol dynamically 
from Berry, but I am completely confused how to convert them in Berry's escape 
sequence as per https://berry.readthedocs.io/en/latest/source/en/Chapter-1.html .

So how I can convert for example typicon E11D to Berry's escape sequence? 

Please give me an example what conversion must be done, so I can make it myself and don't bother you.

Thank you in advance.

**Comment by sfromis** [comment, marked as answer]
What you need is not so much a Berry escape sequence, but instead converting from UTF-16 to UTF-8 encoding. In this case, that could be `p1b2.text='\xEE\x84\x9D'` (3 bytes with hex escapes)

If you do not love bit-twiddling, you could also do the conversion using a bit of Berry code, something like this:
```
def int16_to_utf8(codePoint)
  var b = bytes(-3)
  b.set(0, 0xe0 | (codePoint >> 12 & 0x0f))
  b.set(1, 0x80 | (codePoint >> 6 & 0x3f))
  b.set(2, 0x80 | (codePoint & 0x3f))
  return b.asstring()
end
p1b2.text=int16_to_utf8(0xE11D)
```

**Reply by valdus2** [reply]
Really Grateful...Respect!

**Reply by sfromis** [reply]
With latest dev, you can now use normal Unicode escapes in Berry, like `p1b2.text = '\uE11D'`

**Reply by valdus2** [reply]
Even simpler...Thank you!
